### PR TITLE
#276 JSON attributes as attachments

### PR DESCRIPTION
**Pull request by lmatayoshi**
Changes in the server (in case they're necessary):
https://github.com/instedd/maap-store/pull/73

### CODE REVIEW

File: app/actions/electronicPharmacyStockRecords.js [hunk 1]
```diff
@@ -50,10 +53,13 @@ export const uploadNewElectronicPharmacyStockRecords = () => async (
     const body = new FormData();
     const contents = fs.readFileSync(electronicPharmacyStockRecord.filePath);
     const blob = new Blob([contents]);
+    const electronicPharmacyValues = electronicPharmacyStockRecord.dataValues;
     const mapper = await uploadMapper(
-      electronicPharmacyStockRecord.dataValues,
+      electronicPharmacyValues,
```
[the remark this answers is not in the document]
Comment (pyritewolf) [reply app/actions/electronicPharmacyStockRecords.js:58]: Wait, I just thought this through and understood what irks me - `uploadMapper` makes no sense! it's just a call to `snakeCaseKeys`! can we maybe avoid it?

PS. all this applies to `LabRecords` too

File: app/actions/labRecords.js [hunk 2]
```diff
@@ -113,6 +127,13 @@ export const fetchLabRecord = labRecordId => async (dispatch, getState) => {
   return labRecord;
 };
 
+export const JSONFile = (content, filenme) => {
+  const blob = new Blob([JSON.stringify(content)], {
+    type: 'application/json'
+  });
+  return new File([blob], filenme);
+};
```
Comment (pyritewolf) [inline app/actions/labRecords.js:135]: Food for thought, not a request for change - could this be another `utils` method?

Comment (lmatayoshi) [reply app/actions/labRecords.js:135]: Yep. I think the same but have been lazy about it. I didn't find the proper `utils` file so I had to create a new one. But let's do it!

Comment (lmatayoshi) [reply app/actions/labRecords.js:135]: What do you think about renaming `isDate.js` file to a new more general name such as `utils.js` and keep both methods there?

Comment (pyritewolf) [reply app/actions/labRecords.js:135]: I think from the folder structure that you're meant to make a new file within the `utils` folder for every 'topic' you're addressing with your subset of methods.

From my experience in projects, you often either have one `utils` file (if you have few methods) or a `utils` folder with several, differently-named files for each subset of logics you're covering!

Comment (lmatayoshi) [reply app/actions/labRecords.js:135]: Good to know, thanks!

File: app/actions/electronicPharmacyStockRecords.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lmatayoshi) [reply app/actions/electronicPharmacyStockRecords.js:58]: Even that `uploadMapper` isn't actually doing so much, it tries to represent the mapping between client representation and server parameters for a given entity. It's kind of emphasizing that whenever something is uploaded to the server a potential mapping should be considered. At this moment, almost every entity has its own mapping. By now I'm leaving it the current way, but I'll keep it in mind, thanks!

**Pull request merged**